### PR TITLE
Add .env file handling for local environment variables

### PR DESCRIPTION
**Pull request by basiliskus**
Add .env file handling for local environment variables

## Issue

#487 

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [x] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/Environment.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package gov.hhs.cdc.trustedintermediary.context;
+
+public class Environment {
+
+    private Environment() {}
+
+    public static String getProperty(String key) {
+        if (DotEnv.isNull()) {
```
Comment (jcrichlake) [inline shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/Environment.java:8]: Will DotEnv ever be null? It's getting initialized in the DotEnv class on instantiation

Comment (basiliskus) [reply shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/Environment.java:8]: Good point. The .env file may not exist, but in that case the library will throw an exception instead of returning null, so we need to handle that

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/DotEnv.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package gov.hhs.cdc.trustedintermediary.context;
+
+import io.github.cdimascio.dotenv.Dotenv;
+
+class DotEnv {
```
Comment (halprin) [inline shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/DotEnv.java:5]: Let's add some class level JavaDocs.

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/Environment.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package gov.hhs.cdc.trustedintermediary.context;
+
+public class Environment {
```
Comment (halprin) [inline shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/Environment.java:3]: Let's add some class level JavaDocs.

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/Environment.java (hunk 1 above)
Comment (jorg3lopez) [reply shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/Environment.java]: We opted to not continue with the Environment class as the Dotenv library is doing some of that functionality behind the scenes.

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/DotEnv.java (hunk 2 above)
Comment (jorg3lopez) [reply shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/DotEnv.java:21]: Excellent!

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/DotEnv.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * <p>The <code>load</code> method is primarily intended for testing purposes, allowing for custom
+ * Dotenv instances to be set. It can be useful when configuring specific environments for testing
+ * scenarios.
+ *
+ * @see <a href="https://github.com/cdimascio/dotenv-java">Dotenv Java GitHub Repository</a>
+ * @see io.github.cdimascio.dotenv.Dotenv
+ */
+class DotEnv {
+    private static Dotenv DOTENV;
```
Comment (jorg3lopez) [inline shared/src/main/java/gov/hhs/cdc/trustedintermediary/context/DotEnv.java:22]: This is no longer a final var, any thoughts on renaming it?

File: app/src/test/resources/.env.test [hunk 5]
```diff
@@ -0,0 +1,3 @@
+ENV=local
```
Comment (jcrichlake) [inline app/src/test/resources/.env.test:1]: Is this file needed? I don't see it being loaded in the tests anywhere.

File: generate_env.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Define the content of the .env file
+env_content="ENV=local\nKEY_VAULT-NAME=ti-key-vault-name\nREPORT_STREAM_URL_PREFIX=http://localhost:7071"
```
Comment (jcrichlake) [inline generate_env.sh:4]: Do we need a .env.example as well as a script to generate it?

Comment (jorg3lopez) [reply generate_env.sh:4]: Good catch. The .env.example file was made before the script. It is no longer needed.

File: app/src/test/resources/.env.test (hunk 5 above)
Comment (jorg3lopez) [reply app/src/test/resources/.env.test]: File got deleted. Good one!

File: README.md [hunk 7]
```diff
@@ -32,6 +32,16 @@ The additional requirements needed to contribute towards development are...
...
+### Generating .env File
+
+To set up the necessary environment variables, you can use the `generate_env.sh` script. This script will create a `.env` file in the resource folder with the required configuration. Follow these steps:
+
+1. Navigate to the project directory.
+
+2. Run the `generate_env.sh` script:
+
+   ```bash
+   ./generate_env.sh
```
Comment (halprin) [inline README.md:44]: This is code block isn't closed.

**Review by halprin — CHANGES_REQUESTED** [review]
Just a small thing in the README file.  Other than that, this looks good.

File: README.md (hunk 7 above)
Comment (jorg3lopez) [reply README.md:44]: Good catch, just added the closing.

**Review by halprin — APPROVED** [review]
Nice!
